### ISSUE TITLE
Mobile VO calls bullet point "unpronounceable"

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/molecules-and-light#351]
**Device**
iPad
**OS**
iPadOS 13
**Browser**
Safari
**Problem Description**
For https://github.com/phetsims/QA/issues/503
When Mobile VO is reading through the Light Spectrum dialog, specifically the ranges for the wavelengths and frequencies, it says the white sub bullet points are unpronounceable. It specifically says "unpronounceable, white bullet" before reading out the info. No info is lost, but it may confuse a user into thinking something was missed.

**Visual**
![unpronouncable](https://user-images.githubusercontent.com/41024075/83170903-80a38d80-a0d2-11ea-8657-809a4d9354c0.png)

**Comment by terracoda** [comment]
@KatieWoe, I hear "unpronouncable", too, on mobile and I hear "white bullet" on MacBook Pro. 

Unfortunately, removing the visual bullets with CSS removes the list semantics all together in Safari and then learners using VO lose list semantics and the list count. This is a Safari issue (https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html)

@jessegreenberg, we can discuss testing the work around in the article at MAL's design meeting.

**Comment by jessegreenberg** [comment]
Discussed during today's meeting - 

We are going to see what iOS VO says for a nested list in a vanilla HTML page.
Then we are going to see what adding the aria-role 'list' does to the output.
Then if that doesn't fix it we can see if the CSS has an impact on making it "unpronouncable".

**Comment by jessegreenberg** [comment]
I was able to verify that this is happening in this HTML page as well, which verifies that this is not happening because of our PDOM CSS
<details>

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>TEST PAGE</title>
</head>
<body>

<div>
  <ul>
    <li ><span>Radio, large range:</span>
      <ul>
        <li>Frequencies less than 10 to the 4 Hertz to 10 to the 9 Hertz.</li>
        <li>Wavelengths greater than 10 to the 4 meters to 5 times 10 to the negative 1 meters.</li>
      </ul>
    </li>
  </ul>
</div>

</body>
</html>
```

**Comment by jessegreenberg** [comment]
OK in this example I am following https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html - effectively this is removing semantics with `list-style: none`, and then adding our own back with role=list

```html
<ul role="list">
    <li role="listitem">List item one</li>
    <li role="listitem">List item two with subitems:
        <ul role="list">
            <li role="listitem" class="hidden-list">Subitem 1</li>
            <li role="listitem" class="hidden-list">Subitem 2</li>
        </ul>
    </li>
    <li role="listitem">Final list item</li>
</ul>
```

We still don't hear "white bullet", but mobile VoiceOver says "list start" when it gets to the first nested bullet item.

**Comment by terracoda** [comment]
Ok, I was just testing some code, too.

**Comment by jessegreenberg** [comment]
This sounds OK on desktop VoiceOver too. "White Bullet" is gone, but instead VoiceOver says "list 2 items level 2" when we reach the nested list
```html
<ul>
    <li>Pizza</li>
    <li>Toppings:
        <ul role="list" class="hidden-list">
            <li role="listitem">Mushroom</li>
            <li role="listitem">Onion</li>
        </ul>
    </li>
    <li>Ice Cream</li>
</ul>
```

**Comment by jessegreenberg** [comment]
I just tested with NVDA and what is mentioned in https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html also applies to that AT as well. the CSS removes semantics for NVDA too. Reading from the top with NVDA, the example in https://github.com/phetsims/molecules-and-light/issues/351#issuecomment-640927194 sounds like

"Bullet, Pizza. Bullet, Toppings. List with two items, Mushroom. Onion. Out of list, Bullet, Ice cream."

**Comment by terracoda** [comment]
This is wonderful news, @jessegreenberg. 
I was just verifying this code:
```
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>List Style Test Page</title>
  <style>
	  ul {list-style: none;}
  </style>
</head>
<body>

<div>
<h1>Testing Safari and List Styles</h1>
  <ul role="list">
    <li><span>Radio, large range:</span>
      <ul role="list">
        <li>Frequencies less than 10 to the 4 Hertz to 10 to the 9 Hertz.</li>
        <li>Wavelengths greater than 10 to the 4 meters to 5 times 10 to the negative 1 meters.</li>
      </ul>
    </li>
	<li><span>Microwave, medium range:</span>
      <ul role="list">
        <li>Frequencies.</li>
        <li>Wavelengths.</li>
      </ul>
    </li>
	<li><span>Microwave, medium range:</span>
      <ul role="list">
        <li>Frequencies.</li>
        <li>Wavelengths.</li>
      </ul>
    </li>
  </ul>
</div>

</body>
</html>
```

And I think it is good on my Mac, but I couldn't test in on mobile.

**Comment by terracoda** [comment]
@jessegreenberg, in Safari on MacOS, I hear the list semantics, too, and no more "bullet" on main list items or "white bullet" on sub-list list items.

If all sounds good on JAWS and Chrome, I think this should be a new common code pattern for us. We do not miss-use lists.

**Comment by terracoda** [comment]
@jessegreenberg, I don't think you need the explicit role on the list items, only lists, like this:

```html
<ul role="list">
``` 

and then the CSS somewhere convenient

```css 
ul {list-style: none;}
```

**Comment by jessegreenberg** [comment]
I just tried with JAWS and it sounds reasonable there. Reading from the top:
"Bullet, Pizza. Bullet, Toppings. List with two items, nesting level 1. Mushroom. Onion. List end nesting level 1. Bullet, Ice Cream."

**Comment by jessegreenberg** [comment]
Re https://github.com/phetsims/molecules-and-light/issues/351#issuecomment-640932018

OK @terracoda sounds good! Seems like we can go ahead and do this in the sim.

**Comment by terracoda** [comment]
Awesome, I saw we run with this - no more "unpronounceable"!

**Comment by terracoda** [comment]
@jessegreenberg, where would we open an issue so this fix could be part of common code and eventually maintenance releases?

**Comment by jessegreenberg** [comment]
Hmm, good question I am trying to think of how to do this best too. Have we used nested lists before? If not, there is no need for this as a maintenance release because this is only sim to have this. And I don't want to remove semantics for all lists in PhET code, just ones that have this problem.

This is also very much an Apple bug and we should submit a bug report for it. I am wondering if we should just add this workaround to sim code, and hope this may get resolved by Apple before we need to use nested lists again?

**Comment by terracoda** [comment]
@jessegreenberg, it's not just the "unpronouncable".
I have actually been wondering for a long time, how to get rid of the undesireable voicing of "bullet" and  sometime "bullet, bullet" that VoiceOver says on all list items. 

We do not use nested lists very often, and in a couple of cases I decided against it because the non-visual experience sounded unpleasant to me.

We do use lists (legitimately) often, and think this fix makes lists (nested or not) sound better.

**Comment by terracoda** [comment]
Let's fix MAL and open a general issue elsewhere for discussion.

**Comment by terracoda** [comment]
And totally I agree this is an Apple bug.

**Comment by jessegreenberg** [comment]
OK sounds good, I started https://github.com/phetsims/a11y-research/issues/158 to discuss best markup for unordered lists. In the meantime Ill proceed with a sim specific fix.

**Comment by jessegreenberg** [comment]
Workaround applied, @terracoda  can you confirm this is fixed?

**Comment by terracoda** [comment]
@jessegreenberg, on MacOS, applying the fix to the nested `ul` does remove "white bullet" from the sub-listed items. 

Presumably, this removes "unpronounceable" from the sub-listed items in the mobile experience. I need a dev link to test that, though.

However, I think we will improve the listening experience even more by applying this fix to both the parent `ul` and the nested `ul` and potentially to the `ul` that contain the radio buttons.

I currently hear "bullet, bullet, item 2 of 7" on the main list items, and on the count, "2 of 2" on the  sub-list items in Light Spectrum Diagram dialog.

With the fix applied to both levels of the lists (see screen shot code), I just get the proper counting of the main list items, "2 of 7" and proper counting of the sub-list items, "2 of 2". I don't hear the extraneous "bullet, bullet" not "white bullet".

To me this sounds really good, and I think we should apply the fix more radically, and have QA hammer away at it in RC to see if it causes any undesirable side-effects.
![list-style-code-to-remove-bullet-and-return-list-semantics](https://user-images.githubusercontent.com/1902116/84144902-98b0d080-aa33-11ea-8899-d3a2cd787cba.png)

I'll open a new issue for the `ul`'s that contain the radio buttons, as that `ul` currently has `role="radiogroup"` which may be doing nothing useful. I don't actually hear the role when using VoiceOver.

**Comment by terracoda** [comment]
For this issue, please apply the fix to the parent list in the Light Spectrum Diagram dialog.
```html
<ul role="list" style="list-style: none;">
    <li><span>Radio, large range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies...</li>
        <li>Wavelengths...</li>
      </ul>
    </li>
	<li><span>Microwave, medium range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies.</li>
        <li>Wavelengths.</li>
      </ul>
    </li>
	<li><span>Microwave, medium range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies...</li>
        <li>Wavelengths...</li>
      </ul>
    </li>
    <li><span>Visible, tiny range:</span>
      <ul role="list" style="list-style: none;">
		<li>Shown as a rainbow...</li>
        <li>Frequencies...</li>
        <li>Wavelengths...</li>
      </ul>
    </li>
	<li><span>Ultraviolet, small range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies.</li>
        <li>Wavelengths.</li>
      </ul>
    </li>
	<li><span>X-ray, medium range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies...</li>
        <li>Wavelengths...</li>
      </ul>
    </li>
	<li><span>Gamma ray, medium range:</span>
      <ul role="list" style="list-style: none;">
        <li>Frequencies...</li>
        <li>Wavelengths...</li>
      </ul>
    </li>
  </ul>
```
This results in much less "code/structure verbosity". I do not hear the count on the first item, but rather only on subsequent items, which is kind of nice. Reading backwards through the list creates a miss-read on the fist, i.e., last sub-list item, but I think this is acceptable and it quickly fixes itself on the next nested bullet.

**Clean count on subsequent main list items:**
![clean-count-on-main-list-items](https://user-images.githubusercontent.com/1902116/84146835-e24eea80-aa36-11ea-9e97-e86622263e69.png)

**Clean count on subsequent nested list items** 
Visible light has 3 items, whereas the rest of the wavelengths only have 2 items.
![clean-count-on-nested-list0items](https://user-images.githubusercontent.com/1902116/84146849-e7ac3500-aa36-11ea-9d41-b2eb605b0141.png)

@jessegreenberg, without the fix on the parent list, in the sim code, I hear, "bullet, bullet" and list count, "4 of 7" on the main list items.  
![Screen Shot 2020-06-09 at 10 03 20 AM](https://user-images.githubusercontent.com/1902116/84147772-871df780-aa38-11ea-99ab-e6dbff073430.png)

**Comment by jessegreenberg** [comment]
OK, I added the workaround to the parent list as well in the above commit. @terracoda can you please review? Here is a dev version for mobile:

https://phet-dev.colorado.edu/html/molecules-and-light/1.5.0-dev.51/phet/molecules-and-light_en_phet.html

**Comment by terracoda** [comment]
**Verification in MacOS with Safari  and VO**
1. main list item
<img width="949" alt="verification-main-listitem-macos" src="https://user-images.githubusercontent.com/1902116/84169850-ae35f280-aa53-11ea-9b7f-61c8363e5817.png">
2. sub list item
<img width="951" alt="verification-sublist-item-MacOS" src="https://user-images.githubusercontent.com/1902116/84169859-b1c97980-aa53-11ea-8285-46a6c1339609.png">

**Comment by terracoda** [comment]
On iOS (iphone 7 13.4.1)
I hear no count on the either the main list items or sub-list items, but I get a clear indication of when a list starts and when a list ends.

At the start of the main list I hear:
- Radio, large range, list start

At the start and end of each sub list I hear:
- Frequencies less than ... list start
- Wavelengths ... list end

And I hear for the end of the main list:
- Gamma ray, medium range, list end...then
    - Frequencies... , list start
    - Wavelengths... , list end

Though I don't hear numbered count of the list items (2 of 7, 2 of 2, etc) in the mobile experience, I feel acceptable list semantics are provided and that a clean navigable listening experience is provided.

I like it!

**Comment by terracoda** [comment]
@jessegreenberg, both MacOS 10.13.6 and iOS 13.4.1 are verified.
Closing. Please re-open if I am closing prematurely.
